src/Main.cpp

```diff
@@ -60,10 +60,98 @@ void Main::OnChannelsBox(wxCommandEvent& event) {
 	{
 		DisplayMsg(m);
 	}
+	SendText->Enable(true);
+	SendBtn->Enable(true);
+	SendText->SetFocus();
 }
 
-void Main::DisplayMsg(iMessage& m)
-{
+void Main::DisplayMsg(iMessage& m) {
 	auto msg = Message(m.timestamp, storage.GetCurrentChannel().members[m.member_id].name, wxString(m.text)).FormatToPrint();
 	ChatDisplay->AppendText(msg);
 }
+
+bool Main::PromptLogin(std::string& outUsername) 
+{
+	LoginForm* loginForm = new LoginForm(this);
+	LoginController controller(loginForm, nullptr);
+
+	if (loginForm->ShowModal() == wxID_OK) {
+		outUsername = loginForm->UsernameTextBox->GetValue().ToStdString();
+		delete loginForm;
+		return true;
+	}
+
+	delete loginForm;
+	return false;
+}
+
+void Main::DoLogin()
+{
+	std::string username;
+	if (!PromptLogin(username))
+	{
+		currentUser = User::CreateUser("NoUser");
+		CurrentUserLabel->SetLabelText("User: Not logged in");
+		UpdateLoginButtonLabel();
+		return;
+	}
+
+	auto maybeUser = User::LoadUserByName(username, "../users.txt");
+	if (!maybeUser.has_value())
+	{
+		wxMessageBox("No user with that username exists.", "Login Failed", wxOK | wxICON_ERROR);
+		currentUser = User::CreateUser("NoUser");
```

Comment:
Are you creating a user with name "NoUser"? and if so what purpose does that have